sigopt/hyperopt/base.py

```diff
@@ -0,0 +1,223 @@
+
+  def __getitem__(self, item):
+    return self._trials[item]
+
+  def refresh(self):
+    r = self._trials.refresh()
+    self.do_refresh()
+    return r
+
+  @property
+  def trials(self):
+    return self._trials.trials
+
+  @property
+  def tids(self):
+    return self._trials.tids
+
+  @property
+  def specs(self):
+    return self._trials.specs
+
+  @property
+  def results(self):
+    return self._trials.results
+
+  @property
+  def miscs(self):
+    return self._trials.miscs
+
+  @property
+  def idxs_vals(self):
+    return self._trials.idxs_vals
+
+  @property
+  def idxs(self):
+    return self._trials.idxs
+
+  @property
+  def vals(self):
+    return self._trials.vals
+
+  def assert_valid_trial(self, trial):
+    return self._trials.assert_valid_trial(trial)
+
+  def insert_trial_doc(self, doc):
+    return self._trials.insert_trial_doc(doc)
+
+  def insert_trial_docs(self, docs):
+    return self._trials.insert_trial_docs(docs)
+
+  def new_trial_ids(self, n):
+    return self._trials.new_trial_ids(n)
+
+  def new_trial_docs(self, tids, specs, results, miscs):
+    return self._trials.new_trial_docs(tids, specs, results, miscs)
+
+  def source_trial_docs(self, tids, specs, results, miscs, sources):
+    return self._trials.source_trial_docs(tids, specs, results, miscs, sources)
+
+  def delete_all(self):
+    self.uploaded_tids.clear()
+    return self._trials.delete_all()
+
+  def count_by_state_synced(self, arg, trials=None):
+    return self._trials.count_by_state_synced(arg, trials)
+
+  def count_by_state_unsynced(self, arg):
+    return self._trials.count_by_state_unsynced(arg)
+
+  def losses(self, bandit=None):
+    return self._trials.losses(bandit)
+
+  def statuses(self, bandit=None):
+    return self._trials.statuses(bandit)
+
+  def average_best_error(self, bandit=None):
+    return self._trials.average_best_error(bandit)
+
+  @property
+  def best_trial(self):
+    return self._trials.best_trial
+
+  @property
+  def argmin(self):
+    return self._trials.argmin
+
+  def fmin(
+      self,
+      fn,
+      space,
+      algo=None,
+      max_evals=None,
+      timeout=None,
+      loss_threshold=None,
+      max_queue_len=1,
+      rstate=None,
+      verbose=False,
+      pass_expr_memo_ctrl=None,
+      catch_eval_exceptions=False,
+      return_argmin=True,
+      show_progressbar=True,
+      early_stop_fn=None,
+      trials_save_file="",
+  ):
+    return fmin(
+      fn,
+      space,
+      algo=algo,
+      max_evals=max_evals,
+      timeout=timeout,
+      loss_threshold=loss_threshold,
+      trials=self,
+      rstate=rstate,
+      verbose=verbose,
+      max_queue_len=max_queue_len,
+      allow_trials_fmin=False,  # -- prevent recursion
+      pass_expr_memo_ctrl=pass_expr_memo_ctrl,
+      catch_eval_exceptions=catch_eval_exceptions,
+      return_argmin=return_argmin,
+      show_progressbar=show_progressbar,
+      early_stop_fn=early_stop_fn,
+      trials_save_file=trials_save_file,
+    )
```

Comment:
I think most of this can just be 

```
  def __getattr__(self, name):
    try:
      return getattr(self._trials, name)
    except AttributeError:
      raise AttributeError(f"{type(self.__name__)} object has no attribute {name}")
```

Reply:
It is good! I will try it.

Reply:
done.

Reply:
did you forget to push your changes? I don't see it

Reply:
New code is pushed.